File: src/dataframe/rust/stats/distributions/weibull.test.R
#!/usr/bin/env Rscript

# Weibull distribution test script
# Usage: Rscript weibull.test.R <function> <x_or_p> <shape> <scale> [lower_tail] [log_p]

source("test-helpers.R")

# Define Weibull distribution functions
weibull_density <- function(x, shape, scale, log = FALSE) {
  dweibull(x, shape, scale, log = log)
}

weibull_cumulative <- function(x, shape, scale, lower.tail = TRUE, log.p = FALSE) {
  pweibull(x, shape, scale, lower.tail = lower.tail, log.p = log.p)
}

weibull_quantile <- function(p, shape, scale, lower.tail = TRUE, log.p = FALSE) {
  qweibull(p, shape, scale, lower.tail = lower.tail, log.p = log.p)
}

weibull_random <- function(n, shape, scale) {
  rweibull(n, shape, scale)
}

# Run the test
run_distribution_test(weibull_density, weibull_cumulative, weibull_quantile, weibull_random)

File: src/dataframe/rust/stats/distributions/weibull.test.ts
#!/usr/bin/env -S deno run --allow-read --allow-run

import { callR, printResults, runComparison } from "./test-helpers.ts";

async function main() {
  console.log("Testing R vs Rust Weibull functions...\n");

  const testCases = [
    { func: "dweibull", distribution: "weibull", args: [2, 1, 2, 0] },
    { func: "pweibull", distribution: "weibull", args: [2, 1, 2, 1, 0] },
    {
      func: "qweibull",
      distribution: "weibull",
      args: [0.8646647, 1, 2, 1, 0],
    },
  ];

  const results = await runComparison(testCases);
  printResults(results);

  // Test rweibull separately since it's not available in Rust interface
  console.log("\nR-only tests:");
  try {
    const rweibullResult = await callR("rweibull", "weibull", 1, 1, 2);
    console.log(`rweibull(1, 2) = ${rweibullResult}`);
  } catch (error) {
    console.log(`rweibull: Error - ${(error as Error).message}`);
  }
}

if (import.meta.main) {
  main().catch(console.error);
}
